### ISSUE TITLE
.deb fails with missing icon_16x16x32.png from fpm

### ISSUE THREAD

**Issue by kradalby** [electron-userland/electron-builder#345]
Hi

I am trying to build packages for all systems and my machines runs into a problem with FPM.

To me, it looks like it does not generate the image file icon_16x16x32.png, because i can check the build folder and all the other icon files are generated, but 16x16 is missing.

package.json
```
{
    "name": "p0sX",
    "version": "0.0.1",
    "license": "MIT",
    "main": "main.js",
    "scripts": {
        "postinstall": "install-app-deps",
        "dist": "build",
        "lint": "node ./node_modules/.bin/eslint",
        "lint-jsx": "node ./node_modules/.bin/eslint app/**/**.jsx",
        "electron": "electron ./app",
        "build": "npm run build-main && npm run build-renderer",
        "build-main": "node -r babel-register ./node_modules/webpack/bin/webpack.js --config webpack.config.electron.js --progress --profile --colors",
        "build-renderer": "node -r babel-register ./node_modules/webpack/bin/webpack.js --config webpack.config.production.js --progress --profile --colors",
        "watch-renderer": "node -r babel-register ./node_modules/webpack/bin/webpack.js --config webpack.config.production.js --progress --profile --colors --watch"
    },
    "bin": {
        "electron": "./node_modules/.bin/electron"
    },
    "dependencies": {},
    "devDependencies": {
        "axios": "0.9.1",
        "babel-core": "^6.7.6",
        "babel-loader": "^6.2.4",
        "babel-plugin-add-module-exports": "^0.1.2",
        "babel-plugin-transform-decorators-legacy": "^1.3.4",
        "babel-polyfill": "^6.7.4",
        "babel-preset-es2015": "^6.6.0",
        "babel-preset-react": "^6.5.0",
        "babel-preset-stage-0": "^6.5.0 ",
        "babel-register": "^6.7.2",
        "babel-runtime": "^6.6.1",
        "classnames": "2.2.3",
        "copy-webpack-plugin": "*",
        "del": "*",
        "electron-builder": "^3.6.1",
        "electron-packager": "^7.0.0",
        "electron-prebuilt": "^0.37.6",
        "electron-settings": "^1.0.4",
        "eslint": "*",
        "eslint-config-standard": "^5.1.0",
        "eslint-plugin-promise": "^1.1.0",
        "eslint-plugin-react": "^5.0.1",
        "eslint-plugin-standard": "^1.3.2",
        "fs": "0.0.2",
        "json-loader": "0.5.4",
        "react": "^15.0.1",
        "react-dom": "^15.0.1",
        "react-materialize": "0.14.3",
        "react-redux": "^4.4.5",
        "redux": "^3.4.0",
        "redux-thunk": "2.0.1",
        "webpack": "^1.13.0"
    },
    "build": {
        "app-bundle-id": "org.nuxis.p0sX",
        "productName": "p0sX",
        "iconUrl": "./build/app.ico",
        "directories": {
            "output": "./release"
        },
        "osx": {
            "icon-size": 512,
            "contents": [
                {
                    "x": 425,
                    "y": 200,
                    "type": "link",
                    "path": "/Applications"
                },
                {
                    "x": 120,
                    "y": 200,
                    "type": "file",
                    "path": "computed path to artifact, do not specify it - will be overwritten"
                }
            ]
        }
    }
}
```

package.json
config.json
log of the terminal output
```
$ npm run dist

> p0sX@0.0.1 dist /drone/src/github.com/nuxis/p0sX-client
> build

Installing app dependencies for arch x64 to /drone/src/github.com/nuxis/p0sX-client/app
----------------------------------------------------
Reading icns family from /drone/src/github.com/nuxis/p0sX-client/build/icon.icns...
Extracting icons from /icon.icns...
 Saved 'ic08' element to /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_256x256x32.png.
Extracted 1 images from /icon.icns.

Packaging app for platform linux x64 using electron v0.37.6
{:timestamp=>"2016-04-20T01:34:27.600575+0000", :message=>"Invalid package configuration: Cannot package the path '/drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_16x16x32.png', does it exist?", :level=>:error}
Error: Command failed: fpm -s dir -t deb --architecture amd64 --rpm-os linux --name p0sX --force --after-install /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/after-install --after-remove /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/after-remove --description 

p0sX is a Point of Sales application --maintainer Håvard Lian <kradalby@kradalby.no> --vendor Håvard Lian <kradalby@kradalby.no> --version 0.0.1 --package /drone/src/github.com/nuxis/p0sX-client/dist/p0sX-0.0.1-amd64.deb --deb-compression xz --url https://github.com/nuxis/p0sX --license MIT /drone/src/github.com/nuxis/p0sX-client/dist/p0sX-linux-x64/=/opt/p0sX /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_16x16x32.png=/usr/share/icons/hicolor/16x16/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_24x24x32.png=/usr/share/icons/hicolor/24x24/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_32x32x32.png=/usr/share/icons/hicolor/32x32/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_48x48x32.png=/usr/share/icons/hicolor/48x48/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_64x64x32.png=/usr/share/icons/hicolor/64x64/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_96x96x32.png=/usr/share/icons/hicolor/96x96/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_128x128x32.png=/usr/share/icons/hicolor/128x128/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_256x256x32.png=/usr/share/icons/hicolor/256x256/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/icon_512x512x32.png=/usr/share/icons/hicolor/512x512/apps/p0sX.png /drone/src/github.com/nuxis/p0sX-client/electron-builder-165kgr7MLjmU3AT/p0sX.desktop=/usr/share/applications/p0sX.desktop

   at ChildProcess.exithandler (child_process.js:202:12)
   at emitTwo (events.js:100:13)
   at ChildProcess.emit (events.js:185:7)
   at maybeClose (internal/child_process.js:850:16)
   at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
From previous event:
   at Object.exec (/drone/src/github.com/nuxis/p0sX-client/node_modules/electron-builder/src/util.ts:72:10)
   at LinuxPackager.<anonymous> (/drone/src/github.com/nuxis/p0sX-client/node_modules/electron-builder/src/linuxPackager.ts:201:15)
   at [object Generator].next (native)
From previous event:
   at tsAwaiter (/drone/src/github.com/nuxis/p0sX-client/node_modules/electron-builder/src/awaiter.ts:10:51)
   at Object.build (/drone/src/github.com/nuxis/p0sX-client/node_modules/electron-builder/src/builder.ts:30:59)
   at Object.<anonymous> (/drone/src/github.com/nuxis/p0sX-client/node_modules/electron-builder/src/build-cli.ts:41:2)
   at Module._compile (module.js:413:34)
   at Object.Module._extensions..js (module.js:422:10)
   at Module.load (module.js:357:32)
   at Function.Module._load (module.js:314:12)
   at Function.Module.runMain (module.js:447:10)
   at startup (node.js:146:18)
   at node.js:404:3

npm ERR! Linux 3.16.0-4-amd64
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "dist"
npm ERR! node v5.10.1
npm ERR! npm  v3.8.3
npm ERR! code ELIFECYCLE
npm ERR! p0sX@0.0.1 dist: `build`
npm ERR! Exit status 255
npm ERR! 
npm ERR! Failed at the p0sX@0.0.1 dist script 'build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the p0sX package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs p0sX
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls p0sX
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /drone/src/github.com/nuxis/p0sX-client/npm-debug.log
[info] build failed (exit code 1)
```

node version
```
root@bc05edf1fc6a:/# node --version
v5.10.1
```
npm version
```
root@bc05edf1fc6a:/# npm --version
3.8.3
```

on which system do you want to create installers (OS X, Windows or Linux)
I try to create the installers on Debian 8 Jessie
I have:
wine 1.8 32bit from backports
mono-complete 4.2 from official repo
fpm 1.5 from gem.

I hope I added all the essential information, please let me know if more is needed.

**Comment by develar** [comment]
See https://github.com/electron-userland/electron-builder/issues/320#issuecomment-210750037

Do you have 512x512 in your icns?

**Comment by kradalby** [comment]
My icns is 256x256, 
I have both libopenjpeg5 and libjasper1 installed.
It worked however when I pregenerated all icons, as you did in your app: 
https://github.com/develar/onshape-desktop-shell/tree/master/build/icons

**Comment by develar** [comment]
Fixed in 3.8.0. If you use Travis — please see https://github.com/electron-userland/electron-builder/wiki/Multi-Platform-Build#travis-linux.

16x16 will be generated of missed in icns.

But please note — if your icns contains only 256, but not 512 — it is invalid ICNS. OS X requires 1024, Linux 512.
